### DISCUSSION TITLE
RTSP stream from bridge to VPS

### DISCUSSION THREAD

**Discussion by MRliqiang91** [mrlt8/docker-wyze-bridge#207] [Q&A] [answered]
Hi all,

I've managed to get my RTSP stream to work locally with the bridge (raspberry pi 4) and I am able to view it via VLC on my local network. However, I would like to display the stream over the internet on a webpage. I have a public ip address of my server hosted on AWS.

Does anyone have a good recommendation on how I can do this securely and reliably? Also do I have to change any settings on my router for this to work? 

Brand new at this, so any advice will be very much appreciated

**Comment by mrlt8** [comment, marked as answer]
One way would be to run the bridge in the cloud with `NET_MODE=p2p` and let tutk do all the nat punching.

**Reply by MRliqiang91** [reply]
I did read about the net_mode=p2p. but don't really know how to define the settings for it once I set it to p2p mode.

Is there documentation for this where I can read up on? @mrlt8

**Reply by mrlt8** [reply]
Sorry, not much documentation.

Basically, there are three "modes" that the tutk library can use to connect to the camera: 

- LAN: streams directly from the cam locally
- P2P: streams directly from the cam over your internet
- Relay: relays the stream to/from the cloud over your internet

NET_MODE=p2p allows the bridge to connect via LAN or P2P mode.

Not defining NET_MODE will default to allowing any of the three modes including relay.

**Reply by MRliqiang91** [reply]
I set it to p2p mode and the rtsp stream is viewable from vlc on a local network. How do I access the stream from another network?

**Reply by mrlt8** [reply]
You can use p2p mode to run the bridge on your VPS and have it access the cameras on your local network - like how the app lets you view your cams while away and the camera's tutk library takes care of all the port forwarding and ddns.

Otherwise, you'd need to manually port forward and set your IP.

**Reply by MRliqiang91** [reply]
ah! I get it now. will try that out! thank you!

**Reply by MRliqiang91** [reply]
once the VPS is able to access the cameras.
Do I still use the rtsp stream URL? or is it something else?

**Reply by mrlt8** [reply]
You can use whatever works best for your application. 
HLS should work directly in apple browsers. e.g. `http://my-vps-domain-or-ip:8888/my-cam`

You may also want to disable any unused streams and/or enable authentication.

See the [rtsp-simple-server.yml](https://github.com/aler9/rtsp-simple-server/blob/main/rtsp-simple-server.yml) for a full list of options which can be set using `RTSP_PARAMNAME`, where PARAMNAME is the uppercase name of a parameter you want to set.

**Reply by MRliqiang91** [reply]
@mrlt8 Thanks for the tips! Managed to get it to work and really avoided all the port forwarding and ddns by running the bridge on the server.

Also, great advice to disable the other unused streams.

Appreciate the help in getting this to work!
